### PR TITLE
kokkos-kernels: add variant 'shared'

### PR DESCRIPTION
**Pull request by balay**
This is in sync with `kokkos+shared`

### CODE REVIEW

File: var/spack/repos/builtin/packages/kokkos-kernels/package.py
Comment (sethrj) [inline var/spack/repos/builtin/packages/kokkos-kernels/package.py]:
```suggestion
        options.append(self.define_from_variant('BUILD_SHARED_LIBS', 'shared'))
```

**Review by sethrj — APPROVED** [review]

**Comment by sethrj** [comment]
Happy to let one of the Kokkos maintainers to look at this before I merge.

**Review by lucbv — APPROVED** [review]
This looks good to me, thanks @balay

**Pull request merged**
